src/api/controllers/auth.js

```diff
@@ -26,6 +28,41 @@ const registerController = async (req, res) => {
   return res.status(StatusCodes.CREATED).json({ message: 'User created' });
 };
 
+const forgotPasswordController = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email)
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Email can not be empty' });
+
+  const foundUser = await User.findOne({ email });
+
+  if (!foundUser) throw new NotFound('No user found with this email');
+
+  const token = foundUser.createPasswordResetToken();
+
+  const result = await foundUser.save();
+  const url = `http://localhost:${process.env.PORT}/api/v1/auth/reset-password/${foundUser.email}/${token}`;
```

Comment:
process.env will not work on producyion with the url  you can use this 
const resetUrl = `${req.protocol}://${req.get('host')}/api/v1/users/resetPassword/${resetToken}`;

Reply:
I’ve corrected that.